pkg/apis/pipeline/v1/pipeline_types_test.go

```diff
@@ -668,93 +668,37 @@ func TestPipelineTask_ValidateMatrix(t *testing.T) {
 			Paths:   []string{"matrix.include[0].params[1].name"},
 		},
 	}, {
-		name: "parameters in matrix are strings",
+		name: "parameters in matrix contain references to arrays",
 		pt: &PipelineTask{
 			Name: "task",
+			Params: Params{{
+				Name: "foobar", Value: ParamValue{Type: ParamTypeArray, ArrayVal: []string{"foo", "bar"}},
+			}, {
+				Name: "barfoo", Value: ParamValue{Type: ParamTypeArray, ArrayVal: []string{"bar", "foo"}},
+			}},
 			Matrix: &Matrix{
 				Params: Params{{
-					Name: "foo", Value: ParamValue{Type: ParamTypeString, StringVal: "foo"},
+					Name: "foo", Value: ParamValue{Type: ParamTypeString, StringVal: "$(params.foobar[*])"},
 				}, {
-					Name: "bar", Value: ParamValue{Type: ParamTypeString, StringVal: "bar"},
+					Name: "bar", Value: ParamValue{Type: ParamTypeString, StringVal: "$(params.barfoo[*])"},
 				}}},
 		},
-		wantErrs: &apis.FieldError{
-			Message: "invalid value: parameters of type array only are allowed, but got param type string",
-			Paths:   []string{"matrix.params[foo]", "matrix.params[bar]"},
-		},
 	}, {
-		name: "parameters in matrix are arrays",
+		name: "parameters in matrix contain results references",
```

Comment:
this test is changed in v1, but removed in v1beta1, please sync

Reply:
This is in v1beta1
https://github.com/tektoncd/pipeline/blob/7dedfc68ea9427c49c7b8d767214fce9d4f8bfb2/pkg/apis/pipeline/v1beta1/pipeline_types_test.go#L661